### PR TITLE
Update the DB Schema - Remove User Joins

### PR DESCRIPTION
**Pull request by bhtibrewal**
Date: 

Developer Name: @bhtibrewal 

---

## Issue Ticket Number
#97 
https://github.com/Real-Dev-Squad/skill-tree-backend/issues/118

Follow-up Issue ticket 
https://github.com/Real-Dev-Squad/skill-tree-backend/issues/120

## Description

Since we wanted one source of truth for Users data and no longer wanted Users table in Skill tree removing the DB joins and updated schema
More Details can be found in the issue tickets added below 

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [x] Yes
- [ ] No

### Breaking Changes

- [x] Yes
- [ ] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots

There were no functionality added in this PR so no additional tests added as well 
image

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java [hunk 1]
```diff
@@ -113,18 +110,17 @@ private Page<EndorsementModelFromJSON> createPagedEndorsements(
 
     @Override
     public EndorsementModel createEndorsement(EndorsementDRO endorsementDRO) {
-        UUID userId = endorsementDRO.getUserId();
+        UUID userId = endorsementDRO.getEndorseeId();
         UUID skillId = endorsementDRO.getSkillId();
-        Optional<UserModel> userOptional = userRepository.findById(userId);
+
```
Comment (Shaily20) [inline skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java:115]: Hi @bhtibrewal - Is there any todo here where we check if the user is present by making call to website backend?

Comment (Shaily20) [reply skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java:115]: Also, can't the userId and skillId be empty. Am i missing if we are validating that case somewhere? I see a Notnull annotation on EndorsementDTO class members but that doesn't take care of "" string.

**Comment by Shaily20** [comment]
@bhtibrewal - Will the UserDRO class be removed in future?

File: skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java (hunk 1 above)
Comment (bhtibrewal) [reply skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java:115]: added the follow-up in the description
Yes we verify not null

**Review by iamitprakash — CHANGES_REQUESTED** [review]
update PR desc and better PR title please

File: skill-tree/src/test/java/com/RDS/skilltree/integration/EndorsementsIntegrationTests.java [hunk 2]
```diff
@@ -244,7 +244,7 @@ public void testAPIReturns200_OnEndorsementGivenId() {
                 .body("data.status", anyOf(equalTo("APPROVED"), equalTo("PENDING"), equalTo("REJECTED")))
                 .body("data.endorsementType", anyOf(equalTo("POSITIVE"), equalTo("NEGATIVE")))
                 .body("data.endorsersList", hasSize(1))
-                .body("data.endorsersList[0].endorserId", equalTo("user-2"))
+                .body("data.endorsersList[0].endorseeId", equalTo("user-2"))
```
Comment (iamitprakash) [inline skill-tree/src/test/java/com/RDS/skilltree/integration/EndorsementsIntegrationTests.java:247]: are we testing case where "data.endorsersList" can be null/empty

Comment (bhtibrewal) [reply skill-tree/src/test/java/com/RDS/skilltree/integration/EndorsementsIntegrationTests.java:247]: Yes the lines just above this checks that as well

**Review by iamitprakash — APPROVED** [review]

**Review by Shaily20 — APPROVED** [review]

**Pull request merged**